src/main/groovy/edu/oregonstate/mist/osuevents/db/BindEvent.groovy

```diff
@@ -0,0 +1,71 @@
+import org.skife.jdbi.v2.sqlobject.BinderFactory
+import org.skife.jdbi.v2.sqlobject.BindingAnnotation
+
+import java.lang.annotation.Annotation
+import java.lang.annotation.ElementType
+import java.lang.annotation.Retention
+import java.lang.annotation.RetentionPolicy
+import java.lang.annotation.Target
+
+/**
+ * Custom binder for event object to handle formatting lists as json arrays.
+ */
+@BindingAnnotation(BindEvent.EventBinderFactor.class)
+@Retention(RetentionPolicy.RUNTIME)
+@Target([ElementType.PARAMETER])
+public @interface BindEvent {
+    public static class EventBinderFactor implements BinderFactory {
+        private static ObjectMapper objectMapper = new ObjectMapper()
+
+        /**
+         * Format a list of strings as a json array of strings.
+         * @param list
+         * @return
+         */
+        private static String formatJsonList(List<String> list) {
+            objectMapper.writeValueAsString(list)
+        }
+
+        public Binder build(Annotation annotation) {
+            new Binder<BindEvent, Event>() {
+                public void bind(SQLStatement q, BindEvent bind, Event event) {
+                    q.bind("eventID", event.eventID)
+                    q.bind("title", event.title)
+                    q.bind("description", event.description)
+                    q.bind("locationID", event.locationID)
+                    q.bind("otherLocationName", event.otherLocationName)
+                    q.bind("room", event.room)
+                    q.bind("address", event.address)
+                    q.bind("city", event.city)
+                    q.bind("state", event.state)
+                    q.bind("countyIDs", formatJsonList(event.countyIDs))
+                    q.bind("campusID", event.campusID)
+                    q.bind("eventURL", event.eventURL)
+                    q.bind("photoURL", event.photoURL)
+                    q.bind("ticketURL", event.ticketURL)
+                    q.bind("ticketCost", event.ticketCost)
+                    q.bind("hashtag", event.hashtag)
+                    q.bind("keywords", formatJsonList(event.keywords))
+                    q.bind("tags", formatJsonList(event.tags))
+                    q.bind("allowsReviews", event.allowsReviews)
+                    q.bind("allowUserActivity", event.allowUserActivity)
+                    q.bind("departmentIDs", formatJsonList(event.departmentIDs))
+                    q.bind("contactName", event.contactName)
+                    q.bind("contactEmail", event.contactEmail)
+                    q.bind("contactPhone", event.contactPhone)
+                    q.bind("eventTypeIDs", formatJsonList(event.eventTypeIDs))
+                    q.bind("eventTopicIDs", formatJsonList(event.eventTopicIDs))
+                    q.bind("audienceIDs", formatJsonList(event.audienceIDs))
+                    q.bind("visibility", event.visibility)
+                    q.bind("owner", event.owner)
+                }
```

Comment:
Just a recommendation, you could use the `with` clause in Groovy to simplify the code: https://dzone.com/articles/getting-groovy-with-with.

Reply:
Good suggestion, but I wasn't sure how to do this here because it would need to look something like 
```
q.with {
    bind "eventID", event.eventID
    //bind more stuff
}
```

However, `bind` still refers to a variable being passed to the method, so it thinks I'm reffering to the variable and not the `SQLStatement` method. Any ideas?

Reply:
I followed these instructions for custom bindbeans http://jdbi.org/jdbi2/sql_object_api_argument_binding/